### DISCUSSION TITLE
[General] Any plans for withdraw() support for a different dest address?

### DISCUSSION THREAD

**Discussion by goodacres** [zkSync-Community-Hub/zksync-developers#511] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

We are building an NFT project that will donate proceeds to charity via [The Giving Block](https://thegivingblock.com/). Unfortunately, TGB are not yet setup to handle L2 donations. Subsequently, funds must be bridged back to L1 before donation.

We were hoping to handle donations in the contract for a fully trustless setup, however from reading the docs on withdrawal, we are under the impression that withdraw can only handle transactions back to the caller address on Ethereum L1 .

I was just wondering if there is a specific design decision in regards to why withdrawals cannot be sent to a different address on L1 to the caller on L2?

Thanks in advance,
Jack

### What is the specific issue or error you're encountering?

We believe withdraw() does not allow withdrawal to a different L1 address.

### Can you share the error messages or logs you're receiving, if any?

NA

### Have you made any recent changes to the contract before encountering this issue?

NA

### Are there any external libraries or contracts that your contract interacts with?

NA

### Can you provide the relevant portions of your contract code where the issue is occurring?

NA

### Have you tried to isolate the problem, and if so, what were the results?

NA

### What steps have you already taken to try to resolve the issue?

NA

### Repo Link (Optional)

https://github.com/HumptyDumptyDevs/zkWilly

**Comment by githubdoramon** [comment, marked as answer]
You can withdraw to a different address. Check here - https://docs.zksync.io/build/sdks/js/zksync-ethers/accounts.html#withdraw

**Reply by goodacres** [reply]
Ah my mistake, apologies! Thanks so much for the quick response @githubdoramon. 

I see these docs are for ethers integration - I was wondering if you know of any docs where/if this can be implemented within a contract?

**Reply by githubdoramon** [reply]
The interface for the bridge contract has a l1 destination address - https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/l2-contracts/contracts/bridge/interfaces/IL2Bridge.sol#L29

So if you call it directly on the contract, you can just pass the right address there. Let me know if this answers you question.

**Reply by goodacres** [reply]
This is perfect - thanks again @githubdoramon & apologies for my incorrect assumption!

**Reply by githubdoramon** [reply]
No worries, happy to help

**Reply by goodacres** [reply]
Hey @githubdoramon - sorry to bring this back up! We are just working on implementing the call to withdraw in our contract, but we got an error in the function call (https://sepolia.explorer.zksync.io/tx/0x162184f382950a3b626a15f2489e79c161f5c5b4d56098606e2bd365fd5738df#overview):

```Error function_selector = 0x, data = 0x```

After looking into it, I believe this could be because the Bridge address we have provided may be incorrect..

We are testing on zkSync Sepolia currently, and used the L2ERC20Bridge address found here: https://docs.zksync.io/build/quick-start/useful-address.html#sepolia-token-bridge-contract-addresses

I am just wondering if this correct, as when I do a bridge from portal.zksync.io I notice that the address appears as the default ETH address for the network:

<img width="945" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/48926225/c3870fd9-082b-4703-8d90-862d69a9a0d6">

Thanks in advance!

**Reply by goodacres** [reply]
Just to add, I also checked the address from calling: https://docs.zksync.io/build/api.html#zks-getbridgecontracts

```{"jsonrpc":"2.0","result":{"l1Erc20DefaultBridge":"0x2ae09702f77a4940621572fbcdae2382d44a2cba","l1SharedDefaultBridge":null,"l1WethBridge":"0x0000000000000000000000000000000000000000","l2Erc20DefaultBridge":"0x681a1afdc2e06776816386500d2d461a6c96cb45","l2SharedDefaultBridge":"0x681a1afdc2e06776816386500d2d461a6c96cb45","l2WethBridge":"0x0000000000000000000000000000000000000000"},"id":1}```

**Reply by goodacres** [reply]
Adding that I also tried the `l2WethBridge` address of `0x0000000000000000000000000000000000000000` with the same error: https://sepolia.explorer.zksync.io/tx/0xf92373ddef28d866339dfa456f57db40bcdca00a63c738def6007729a0acc39b

**Reply by goodacres** [reply]
Will spend some time debugging with https://docs.zksync.io/build/test-and-debug/era-test-node.html and report back
